### ISSUE TITLE
[Bug]: Illegal instructions when run milvus on CPU with only SSE42 support.

### ISSUE THREAD

**Issue by matrixji** [milvus-io/milvus#12335]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster): standalone 
- SDK version(e.g. pymilvus v2.0.0rc2): N/A
- OS(Ubuntu or CentOS): ubuntu 18.04 and macOS 12.0.1
- CPU/Memory: KVM emulated (support SSE42, without AVX2), M1
- GPU: N/A
- Others:
```

### Current Behavior

I see some discussion on slack, some guys want to run standalone mode with docker-compose on MacBook(M1) it fails due to Illegal instructions Error.

As debugging under docker on M1 is pretty difficult, I use a virtual machine with a simulated SSE42 only CPU to reproduce it.

Finally, I see 2 root causes for this error(also exists in version 1.x branch).
- While building faiss, we are using -mf16c this compile flag will use some AVX2 instructions.
- In the file `distances_simd_avx.cpp` , there is a global variable `lookup` that will be initialized when the program started, and it will call _mm256_setr_epi8() which require AVX2 instructions also.

After fixing the above bugs, I've tested hello milvus success under  M1 MacBook.
I'll PR it later.

### Expected Behavior

Accroding the documentation: https://milvus.io/docs/v2.0.0/operational_faq.md#Why-does-Milvus-return-illegal-instruction-during-startup
We could see milvus need `at least one of these`, so we need provide milvus running under SSE42 only CPU.

### Steps To Reproduce

```markdown
see above.
```

### Anything else?

This fix may help for the below issues: #12296 #7989 #6722

**Comment by matrixji** [comment]
> Is it possible for this fix to be backported to `1.1.1`?
> 
> Unfortunately I have a hard dependency from a third-party on Milvus `1.1.1` but this fix appears to have only gone into `2.0`.

The PR is ongoing on 1.1 branch.  https://github.com/milvus-io/milvus/pull/17143
If you just want to have a try 1.1.1 CPU images on MacBook M1,  you could try using image with `matrixji/milvus:1.1-enhanced-cpu-9408a4e`

```shell
❯ docker run --rm  --platform linux/amd64 matrixji/milvus:1.1-enhanced-cpu-9408a4e
Unable to find image 'matrixji/milvus:1.1-enhanced-cpu-9408a4e' locally
1.1-enhanced-cpu-9408a4e: Pulling from matrixji/milvus
2d473b07cdd5: Pull complete
26ab006df284: Pull complete
06fe6866dee7: Pull complete
e256fcdc1b3d: Pull complete
Digest: sha256:5f04b0d8ca42331b9a4d71d79a42afcc4df1c8109668151ee7da86b59d62f845
Status: Downloaded newer image for matrixji/milvus:1.1-enhanced-cpu-9408a4e

    __  _________ _   ____  ______
   /  |/  /  _/ /| | / / / / / __/
  / /|_/ // // /_| |/ / /_/ /\ \
 /_/  /_/___/____/___/\____/___/

Welcome to use Milvus!
Milvus Release version: v1.1.1, built at 2022-05-09 06:37.51, with OpenBLAS library.
You are using Milvus CPU edition
Last commit id: 9408a4e26e01f2073e6d82f1b95961913e5df957

Loading configuration from: /var/lib/milvus/conf/server_config.yaml
NOTICE: You are using SQLite as the meta data management. We recommend change it to MySQL.
Supported CPU instruction sets: sse4_2
FAISS hook SSE42
Milvus server started successfully!
```

Ref: https://jibin.net/2022/05/some-enhancement-of-milvus-1.1.html  Sorry for I've written this article in Chinese, and it should be easy to read through Google translate.

**Comment by miguelsmuller** [comment]
following this thread
